e2-studio-star-rx72n-firmware/src/rx_stack_monitor.c

```
@@ -0,0 +1,323 @@
+ *
+ * @par NASA Power of 10 Compliance:
+ * - Rule 2: ✓ Loop bound = tx_thread_stack_size (statically known at entry)
+ * - Rule 5: ✓ 2 preconditions (NULL checks), 2 postconditions
+ */
+rx_err_t rx_stack_monitor_get_free_bytes(const TX_THREAD* thread_ptr, uint32_t* free_bytes)
+{
+    /* Precondition: null pointer checks */
+    RX_CHECK_NULL_PTR(thread_ptr, s_tag, "thread_ptr must not be NULL");
+    RX_CHECK_NULL_PTR(free_bytes, s_tag, "free_bytes must not be NULL");
+
+    const uint8_t* const stack_base = (const uint8_t*)thread_ptr->tx_thread_stack_start;
```

Clang (14.0.6)

[warning] variable 'stack_base' is not initialized

(cppcoreguidelines-init-variables)